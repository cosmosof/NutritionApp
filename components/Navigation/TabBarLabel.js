// @flow

import React from 'react';
import {Text} from 'react-native';
import {Colors} from '../../Constants';

import styles from './Styles/TabBarLabelStyles';

type Props = {
  focused: boolean,
  title: string,
  textStyle?: Object,
  styles?: Object,
};

export default function TabBarLabel(props: Props) {
  const {textStyle, title} = props;
  const {text} = styles;

  return (
    <Text style={[text, textStyle, {color: props.focused ? Colors.vividBlue : Colors.tabIconDefault}]}>{title}</Text>
  );
}

TabBarLabel.defaultProps = {
  styles: undefined,
  textStyle: undefined,
};
