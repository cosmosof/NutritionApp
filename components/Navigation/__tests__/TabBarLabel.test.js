import React from 'react';
import {StyleSheet} from 'react-native';
import {shallow} from 'enzyme';
import TabBarLabel from '../TabBarLabel';
import {TestHelpers} from '../../../Constants';

const {sampleText} = TestHelpers;

describe('<TabBarLabel>', () => {
  const Component = (<TabBarLabel
    focused={false}
    title={sampleText}
  />);

  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external text style from object', () => {
      const Clone = React.cloneElement(Component, {styles: {textAlign: 'center'}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external text style from ID', () => {
      const styles = StyleSheet.create({
        containerStyle: {
          padding: 5,
        },
      });
      const Clone = React.cloneElement(Component, {styles: styles.containerStyle});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external container style from array', () => {
      const style1 = {backgroundColor: 'yellow'};
      const styles = StyleSheet.create({
        style2: {
          maxHeight: 32,
        },
      });
      const Clone = React.cloneElement(Component, {styles: [style1, styles.style2]});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external text style from object', () => {
      const Clone = React.cloneElement(Component, {textStyle: {color: 'black'}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external text style from ID', () => {
      const styles = StyleSheet.create({
        textStyle: {
          fontSize: 12,
        },
      });
      const Clone = React.cloneElement(Component, {textStyle: styles.textStyle});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders external text style from array', () => {
      const style1 = {color: 'blue'};
      const styles = StyleSheet.create({
        style2: {
          fontFamily: 'sans-serif',
        },
      });
      const Clone = React.cloneElement(Component, {textStyle: [style1, styles.style2]});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
