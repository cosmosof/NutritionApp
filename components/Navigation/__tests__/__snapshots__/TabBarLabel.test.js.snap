// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBarLabel> Structure renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      undefined,
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external container style from array 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      undefined,
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external text style from ID 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      undefined,
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external text style from ID 2`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      Object {
        "fontSize": 12,
      },
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external text style from array 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      Array [
        Object {
          "color": "blue",
        },
        Object {
          "fontFamily": "sans-serif",
        },
      ],
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external text style from object 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      undefined,
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;

exports[`<TabBarLabel> Structure renders external text style from object 2`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "AvenirNextCondensed-DemiBold",
        "textAlign": "center",
      },
      Object {
        "color": "black",
      },
      Object {
        "color": "#58C6F2",
      },
    ]
  }
>
  Sample text
</Text>
`;
